plugin-k8s/src/lib.rs

```diff
@@ -43,30 +50,128 @@ impl AlumetPlugin for K8sPlugin {
         if !v2_used {
             anyhow::bail!("Cgroups v2 are not being used!");
         }
-        let final_li_metric_file: Vec<CgroupV2MetricFile> = cgroup_v2::list_all_k8s_pods_file("/sys/fs/cgroup/kubepods.slice/")?;
-        let metrics_result = k8s_probe::Metrics::new(alumet);
+        let metrics_result = Metrics::new(alumet);
         let metrics = metrics_result?;
-        let probe = K8SProbe::new(metrics.clone(), final_li_metric_file)?;
-        alumet.add_source(Box::new(probe), TriggerSpec::at_interval(self.config.poll_interval));
+        self.metrics = Some(metrics.clone());
+        let final_li_metric_file: Vec<CgroupV2MetricFile> = cgroup_v2::list_all_k8s_pods_file(&self.config.path)?;
+
+        //Add as a source each pod already present
+        for metric_file in final_li_metric_file {
+            let counter_tmp_tot: CounterDiff = CounterDiff::with_max_value(CGROUP_MAX_TIME_COUNTER);
+            let counter_tmp_usr: CounterDiff = CounterDiff::with_max_value(CGROUP_MAX_TIME_COUNTER);
+            let counter_tmp_sys: CounterDiff = CounterDiff::with_max_value(CGROUP_MAX_TIME_COUNTER);
+            let probe = K8SProbe::new(metrics.clone(), metric_file, counter_tmp_tot, counter_tmp_sys, counter_tmp_usr)?;
+            alumet.add_source(Box::new(probe), TriggerSpec::at_interval(self.config.poll_interval));
+        }
+
         return Ok(());
     }
 
     fn stop(&mut self) -> anyhow::Result<()> {
         Ok(())
     }
+
+    fn post_pipeline_start(&mut self, pipeline: &mut alumet::pipeline::runtime::RunningPipeline) -> anyhow::Result<()> {
+        let control_handle = pipeline.control_handle();
+        let plugin_name = self.name().to_owned();
+
+        // let metrics = self.metrics.clone().unwrap();
+        let metrics = self.metrics.clone().unwrap();
+        let poll_interval = self.config.poll_interval;
+
+        struct PodDetector {
+            plugin_name: String,
+            metrics: Metrics,
+            control_handle: ControlHandle,
+            poll_interval: Duration,
+        }
+        // struct JobDetector {
+        //     config_path: PathBuf,
+        //     time_spend_tot: TypedMetricId<u64>,
+        //     time_spend_usr: TypedMetricId<u64>,
+        //     time_spend_sys: TypedMetricId<u64>,
+        //     control_handle: ControlHandle,
+        //     plugin_name: String,
+        //     poll_interval: Duration,
+        // }
+
+        impl EventHandler for PodDetector {
+            fn handle_event(&mut self, event: Result<Event, notify::Error>) {
+                // Handle_Event: Ok(Event { kind: Create(Folder), paths: ["/sys/fs/cgroup/kubepods.slice/kubepods-besteffort.slice/TESTTTTT"], attr:tracker: None, attr:flag: None, attr:info: None, attr:source: None })
+                // Handle_Event: Ok(Event { kind: Remove(Folder), paths: ["/sys/fs/cgroup/kubepods.slice/kubepods-besteffort.slice/TESTTTTT"], attr:tracker: None, attr:flag: None, attr:info: None, attr:source: None })
```

Comment:
You can add a comment above those two comments to explain what it is.
For instance "the event, if successfull, looks like the following"